sequins.go

```diff
@@ -118,6 +119,53 @@ func (s *sequins) init() error {
 	return nil
 }
 
+func (s *sequins) joinCluster(zkWatcher *zk.Watcher, routableIpAddress string) (*sharding.Peers, error) {
+	lock := zkWatcher.CreateIDAssignmentLock()
```

Comment:
I changed the locking to take out a much larger lock. Previously it just locked when a shardID wasn't provided and only locked until one was found. Now it locks until the cluster has been joined. This avoids the race condition of 2 Sequins nodes starting up, the first one choosing ID `X`, releasing the lock, and then the second one also choosing `X` while the first one is stuck on waiting for the cluster to stabilize [0]

[0] I realize this is very unlikely since the second one would also need to wait for the cluster to stabilize. But if, for some reason, we ever changed this logic or the two `WaitToConverge` times were different then this could be an issue